ruby_programming/automated_testing/introduction.md

```diff
@@ -0,0 +1,52 @@
+### Introduction
+
+Hopefully, you’ve been having fun developing in Ruby so far. Perhaps one thing you may not have been enjoying as much, is the process of manually testing your app to ensure it still works after making any changes. Or even worse running your app, getting far into its execution and discovering you have a bug, and then having to do it all over again to verify that you've fixed the bug.
+
+Automated tests are the alternative to manual testing. You've already seen them in action while doing the [Ruby basic exercises](https://github.com/TheOdinProject/ruby-exercises/tree/main/ruby_basics) earlier in this course. Those exercises tasked you with writing code to make a bunch of existing failing tests pass. Over the next few lessons you will learn how to write those tests yourself and integrate writing them into your workflow.
+
+### Learning Outcomes
+
+- What is automated testing?
+- What is a test suite?
+- What are the benefits of automated testing over manual testing?
+- What is a regression?
+
+#### What are Automated Tests?
+
+Automated tests, as the name suggests, automate the process of testing your projects. As you write a new feature, you also write tests for the feature that verify the correctness of it. If the tests pass, you can be confident that the code works as intended. If they fail you know a mistake has been made that needs to be corrected.
+
+There are different types of automated tests. Some are low level and only test individual classes and methods. These are known as unit tests.
+
+Other tests are high level and test your system as a whole by simulating how a real user would interact with the system. These are known as end-to-end tests. All of your tests put together are what's known as a test suite.
+
+### What Problems do Automated Tests Solve?
+
+#### They Save You Time
+
+The most obvious benefit automated testing has over manual testing is the amount of time it can save you.
+
+There's a lot of repetition involved with manual testing. Every time a new feature is added you have to test the feature as well as existing features to ensure they haven't been broken by the changes you've made.
+
+Depending on the size of the project, it could take anywhere from a few minutes to a few weeks in order to exhaustively test everything manually. This amount of time only increases as the project grows in size and complexity.
+
+On the flip side, automated tests take a fraction of the time to run, often only seconds. And you only need to write an automated test once, after which it will test that behavior every time the test suite is run as opposed to having to repeatedly testing the same thing manually every time a change is made.
```

Comment:
```suggestion
On the flip side, automated tests take a fraction of the time to run, often only seconds. And you only need to write an automated test once, after which it will test that behavior every time the test suite is run as opposed to repeatedly testing the same thing manually every time a change is made.
```

Reply:
Nice catch 🎯  thanks dude.